projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -845,37 +911,23 @@ export class IgxDateRangePickerComponent extends DisplayDensityBase
         }
     }
 
+    private initialSetValue() {
+        // if there is no value, no ngControl but we have inputs we may have value set trough
+        // inputs' ngModels - we should generate our initial control value
+        if (!this.value && this.hasProjectedInputs && !this._ngControl) {
```

Comment:
Likely a candidate for refactor - this is also the condition on which you don't use `updateInputs` (since the value is coming from the projected inputs)